tests/tpg300.py

```diff
@@ -83,3 +87,70 @@ def test_that_GIVEN_a_disconnected_emulator_WHEN_getting_pressure_THEN_INVALID_a
         for channel in CHANNELS:
             pv = "PRESSURE_{}".format(channel)
             self.ca.assert_that_pv_alarm_is(pv, self.ca.Alarms.INVALID)
+
+    def set_switching_function(self, function):
+        self.ca.set_pv_value("FUNCTION", function)
+
+    def set_switching_function_thresholds(self, threshold_low, exponent_low, threshold_high, exponent_high, circuit_assignment):
+        self.ca.set_pv_value("FUNCTION:LOW:SP", threshold_low)
+        self.ca.set_pv_value("FUNCTION:LOW:E:SP", exponent_low)
+        self.ca.set_pv_value("FUNCTION:HIGH:SP", threshold_high)
+        self.ca.set_pv_value("FUNCTION:HIGH:E:SP", exponent_high)
+        self.ca.set_pv_value("FUNCTION:ASSIGN:SP", circuit_assignment)
+        self.ca.process_pv("FUNCTION:ASSIGN:SP:OUT")
+
+    def check_switching_function_thresholds(self, function, thresholds, check_pv_is=True):
+        readback_string = str(thresholds[0]) + 'E' + str(thresholds[1]) + ',' + str(thresholds[2]) + 'E' + \
+                          str(thresholds[3]) + ',' + str(thresholds[4])
+        if check_pv_is:
+            self.ca.assert_that_pv_is("FUNCTION:"+function+":RB", readback_string)
+            self.ca.assert_that_pv_is("FUNCTION:"+function+":LOW:RB", thresholds[0])
+            self.ca.assert_that_pv_is("FUNCTION:"+function+":LOW:E:RB", thresholds[1])
+            self.ca.assert_that_pv_is("FUNCTION:"+function+":HIGH:RB", thresholds[2])
+            self.ca.assert_that_pv_is("FUNCTION:"+function+":HIGH:E:RB", thresholds[3])
+            self.ca.assert_that_pv_is("FUNCTION:"+function+":ASSIGN:RB", thresholds[4])
+        else:
+            self.ca.assert_that_pv_is_not("FUNCTION:"+function+":RB", readback_string)
+            self.ca.assert_that_pv_is_not("FUNCTION:"+function+":LOW:RB", thresholds[0])
```

Comment:
Similar to above, there is an `assert_that_pv_is_not_number` with a tolerance which is more suitable for floating-point values.